storage/cloud-client/hmac_test.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2019 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Tests for hmac.py. Requires GOOGLE_CLOUD_PROJECT (valid project) and
+HMAC_KEY_TEST_SERVICE_ACCOUNT (valid service account email) env variables to be
+set in order to run.
+"""
+
+
+import os
+
+from google.cloud import storage
+import pytest
+
+import hmac
+
+
+PROJECT_ID = os.environ['GOOGLE_CLOUD_PROJECT']
+SERVICE_ACCOUNT_EMAIL = os.environ['HMAC_KEY_TEST_SERVICE_ACCOUNT']
+
+
+@pytest.fixture()
+def new_hmac_key():
+    """
+    Fixture to create a new HMAC key, and to guarantee all keys are deleted at
+    the end of each test.
+    """
+    hmac_key = hmac.create_key(PROJECT_ID, SERVICE_ACCOUNT_EMAIL)
+    yield hmac_key
+    storage_client = storage.Client(project=PROJECT_ID)
+    hmac_key = storage_client.get_hmac_key_metadata(hmac_key.access_id,
```

Comment:
I'm a little confused by this line - is this just updating the metadata in case it's changed in the test? A comment might be helpful to clarify.

Reply:
Yep, that's exactly what it is-- the hmac_key object will be out of date if the state has been changed during the test. Will add comment.